### PR TITLE
DISCL-296: DesktopStreamer improvements

### PR DESCRIPTION
**Pull request by dnachbaur**
- UI file
- remove selection rectangle
- editable list of stream hostnames

### CODE REVIEW

File: apps/DesktopStreamer/MainWindow.cpp [hunk 1]
```diff
@@ -68,136 +70,44 @@ typedef __int32 int32_t;
...
 #define CURSOR_IMAGE_FILE     ":/cursor.png"
 
 MainWindow::MainWindow()
     : _stream( 0 )
-    , _desktopSelectionWindow( new DesktopSelectionWindow( ))
-    , _x( 0 )
-    , _y( 0 )
-    , _width( 0 )
-    , _height( 0 )
+    , _cursor( QImage( CURSOR_IMAGE_FILE ).scaled( 20, 20, Qt::KeepAspectRatio))
```
Comment (rdumusc) [inline apps/DesktopStreamer/MainWindow.cpp]: #define 20, and what about retina displays?

File: apps/DesktopStreamer/MainWindow.cpp [hunk 2]
```diff
@@ -343,43 +231,28 @@ void MainWindow::_processStreamEvents()
...
-#endif
-
-void MainWindow::_shareDesktopUpdate()
-{
     QTime frameTime;
     frameTime.start();
 
-    const QPixmap desktopPixmap =
-        QApplication::primaryScreen()->grabWindow( 0, _x, _y, _width, _height );
+    QPixmap pixmap = QApplication::primaryScreen()->grabWindow( 0 );
```
Comment (rdumusc) [inline apps/DesktopStreamer/MainWindow.cpp]: why not const?

File: apps/DesktopStreamer/MainWindow.h [hunk 3]
```diff
@@ -49,70 +50,32 @@
...
+class MainWindow : public QMainWindow, public Ui::MainWindow
 {
     Q_OBJECT
 
 public:
     MainWindow();
-
-signals:
-    void streaming( bool enabled );
+    ~MainWindow();
```
Comment (rdumusc) [inline apps/DesktopStreamer/MainWindow.h]: is it needed to define it explicitly? Implementation is empty

File: apps/DesktopStreamer/MainWindow.cpp (hunk 1 above)
Comment (dnachbaur) [inline apps/DesktopStreamer/MainWindow.cpp]: dunno, I just moved the code. 20 \* devicePixelRatio()?

**Comment by rdumusc** [comment]
Otherwise +1, code looks fine. Would you mind adding a screenshot of the what the new UI and hostname selection box looks like?

File: apps/DesktopStreamer/MainWindow.cpp (hunk 1 above)
Comment (rdumusc) [inline apps/DesktopStreamer/MainWindow.cpp]: that should do!

**Comment by dnachbaur** [comment]
desktopstreamer layout

**Comment by dnachbaur** [comment]
Updated

**Comment by rdumusc** [comment]
Very good! I asked Pawel about the new hostnames for the 2 new displaywalls, but no reply yet. I don't know if you want to wait to include them in this commit, otherwise you can merge and we do a 0.9.1 release next week

**Comment by rdumusc** [comment]
Just in case, the answer is:
bbpav05 -> 5th floor
bbpav06 -> 6th floor

**Comment by dnachbaur** [comment]
You should really relax the per-commit-is-release policy :)

**Comment by dnachbaur** [comment]
Added the new hosts

**Comment by rdumusc** [comment]
Haha, there is no per-commit-is-release policy :-) I was just thinking we could do everything in one go, that's all!

**Pull request merged**